pkg/doc/signature/verifier/verifier.go

```diff
@@ -71,7 +71,7 @@ func (dv *DocumentVerifier) verifyObject(jsonLdObject map[string]interface{}) er
 	}
 
 	for _, p := range proofs {
-		publicKey, err := dv.pkResolver.Resolve(p.Creator)
+		publicKey, err := dv.pkResolver.Resolve(p.JSONLdObject())
```

Comment:
Here is a place of the main change - we pass a proof map to the `resolver` instead of concrete proof attribute. A resolver impl decides which proof attributes are needed for public key resolution (`creator`, `verificationMethod`, any other).

Reply:
I think you should first look if "verificationMethod" is present. If not then search for creator. If creator is not present, then fail. This will cover both cases (and older VC versions) and client will not have to worry about it.

Reply:
I changed it to the way you proposed @sandrask . Please take a look.